bundles/core/org.eclipse.smarthome.core.transform/src/main/java/org/eclipse/smarthome/core/transform/actions/Transformation.java

```diff
@@ -37,23 +52,31 @@
      * @return the transformed value or the original one, if there was no service registered for the
      *         given type or a transformation exception occurred.
      */
-    public static String transform(String type, String function, String value) {
-        String result;
-        TransformationService service = TransformationHelper
-                .getTransformationService(TransformationActivator.getContext(), type);
+    public static @Nullable String transform(String type, String function, String value) {
         Logger logger = LoggerFactory.getLogger(Transformation.class);
-        if (service != null) {
-            try {
-                result = service.transform(function, value);
-            } catch (TransformationException e) {
-                logger.error("Error executing the transformation '{}': {}", type, e.getMessage());
-                result = value;
-            }
-        } else {
-            logger.warn("No transformation service '{}' could be found.", type);
+        String result;
+        try {
+            result = trans(type, function, value);
+        } catch (TransformationException e) {
+            logger.error("Error executing the transformation '{}': {}", type, e.getMessage());
```

Comment:
You will now log an error, where before was a warning (if the service is not found).
As we actually declare that this function expects failures and treats them internally already, I would claim that we should rather have only DEBUG, INFO or at most WARN level here - I'd prefer DEBUG.